### PR TITLE
fix: [DIOS-7649] Add SOREUSEPORT option to bound TCP sockets to workon BSD systems

### PR DESCRIPTION
**Pull request by bcostdolby**
…on BSD

### CODE REVIEW

File: src/ws/websocketserver.cpp [hunk 1]
```diff
@@ -92,7 +92,16 @@ int WebSocketServer::Run()
 
 	//Set SO_REUSEADDR on a socket to true (1):
 	int optval = 1;
-	setsockopt(server, SOL_SOCKET, SO_REUSEADDR, &optval, sizeof(optval));
+	int result = setsockopt(server, SOL_SOCKET, SO_REUSEADDR, &optval, sizeof(optval));
```
Comment (eck1001) [inline src/ws/websocketserver.cpp:95]: change-request: Lines 94-97 were these suppose to be deleted? They look like duplicates.

**Review by eck1001 — CHANGES_REQUESTED** [review]

File: src/rtmp/rtmpserver.cpp [hunk 2]
```diff
@@ -76,7 +76,20 @@ int RTMPServer::BindServer()
...
 // Ignore coverity error: "this->server" is passed to a parameter that cannot be negative.
 // coverity[negative_returns]
-	(void)setsockopt(server, SOL_SOCKET, SO_REUSEADDR, &optval, sizeof(optval));
+	int result = setsockopt(server, SOL_SOCKET, SO_REUSEADDR, &optval, sizeof(optval));
+	if (result < 0)
+	{
+		Warning("-RTMPServer::BindServer() Failed to set SO_REUSEADDR on socket. reason: %s\n", strerror(errno));
+	}
+
+	// See https://stackoverflow.com/questions/14388706/how-do-so-reuseaddr-and-so-reuseport-differ
```
Comment (eck1001) [inline src/rtmp/rtmpserver.cpp:85]: 🎉: love the context notes -- thanks - BSD sockets... what a world

**Comment by harryz2000** [comment]
lgtm

**Review by harryz2000 — APPROVED** [review]

**Comment by mildsunrise** [comment]
wait, do we want to make medooze work in systems other than Linux? we abandoned that path quite some time ago, and I'm not sure it's a good idea to retake it...

**Comment by harryz2000** [comment]
I think this is for Linux over Docker. @bcostdolby can confirm.

**Comment by bcostdolby** [comment]
> wait, do we want to make medooze work in systems other than Linux? we abandoned that path quite some time ago, and I'm not sure it's a good idea to retake it...

I am just trying to make it work in docker on the m3. It is using rosetta to emulate so is not a arm arch running but intel. It works fine but there are some minor behaviour differences and IMO using SO_REUSEPORT is correct anyway.

**Comment by mildsunrise** [comment]
I am still a bit confused, doesn't docker desktop start a Linux VM and run docker deamon there? why would this change be needed for such a setup, if it's running inside a Linux VM? :o

**Comment by eck1001** [comment]
Docker linux containers still share the host kernel and SO_REUSEPORT is a feature implemented at the kernel level

**Pull request merged**